discourse_rock/patches/rake.patch

```diff
@@ -0,0 +1,87 @@
+--- a/lib/tasks/discourse-charm.rake
++++ b/lib/tasks/discourse-charm.rake
+@@ -0,0 +1,84 @@
++# frozen_string_literal: true
++
++desc "Promote a user to admin"
++task "admin:promote", [:email] => [:environment] do |_, args|
++email = args[:email]
++  if !email || email !~ /@/
++    puts "ERROR: Expecting rake admin:promote[some@email.com]"
++    exit 1
++  end
++  if User.find_by_email(email).nil?
```

Comment:
we are calling `find_by_email` twice. Either put  `admin = User.find_by_email(email)` before the `if` and check for `nil` or do it all together in the if (assign and check not nil).